render.c

```diff
@@ -311,6 +311,10 @@ void render_frame(struct swaylock_surface *surface) {
 			}
 		}
 
+		// Ensure buffer size is multiple of buffer scale - required by protocol
+		new_height += surface->scale - (new_height % surface->scale);
+		new_width += surface->scale - (new_width % surface->scale);
```

Comment:
If `new_width` is a multiple of `surface->scale`, this unnecessarily add `surface->scale` to the size. But this doesn't sound like a big deal, `new_width` isn't used for centering the sub-surface.